lib/bsdlib/bsd_os.c

```diff
@@ -186,9 +238,11 @@ ISR_DIRECT_DECLARE(ipc_proxy_irq_handler)
 	return 1; /* We should check if scheduling decision should be made */
 }
 
+
 ISR_DIRECT_DECLARE(rpc_proxy_irq_handler)
```

Comment:
Can we get the "fd" that the event goes to from here? This should then match the bsd_os_timedwait() "context" input parameter. Waking up all threads does not make much sense if the event is not theirs.

Reply:
I'm afraid this will not be straightforward. From I've seen, `bsd_os_timedwait` is called from multiple places in `bsdlib`, and different calls use different objects/structures as a `context` parameter. Therefore we cannot depend soloely on a socket file descriptor. It's good for `pool`, but not for other use-cases :(